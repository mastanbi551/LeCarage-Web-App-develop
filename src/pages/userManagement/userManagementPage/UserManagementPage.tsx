import React from 'react';
import {
  DynamicDataTable,
  PageHeader,
} from '../../../components';
import './UserManagementPage.scss';

export interface userProps {  
  'CellPhone Number': string,
  'Email Address': string,
  'First Name': string,
  'Prathima': string,
  'Last Name': string,
  'Status': string,
  'User Id': string,
  'User Type': string,
  'Receptionist': string
}
interface UserManagementInterface {
  handleAddUser: () => void;
  columns: { Header: string; accessor: string; }[];
  searchValue: string[],
  onViewHandler: (rowInfo: object) => void;
  onEditHandler: (rowInfo: object) => void;
  onDeleteHandler:  (rowInfo: object) => void;
  users: [];
  filterValues: {mainMenu: string, subMenus: string[]}[];
  errorMessage: string;
  access: string;
  itemsPerPage: number;
  rowsPerPageValueChange: (value: string) => void
}
const UserManagementPage: React.FC<UserManagementInterface> = ({
  handleAddUser,
  columns,
  searchValue,
  onViewHandler,
  onEditHandler,
  onDeleteHandler,
  users,
  filterValues,
  errorMessage,
  access,
  itemsPerPage,
  rowsPerPageValueChange
}) => {
  return (
    <>
      <div className='users-container'>
        <div className='users-container__header-content'>
          <PageHeader
            text='Users'
            name='users-container__header-content__heading'
            isButtonDisable={true}
            buttonName='Add New'
            buttonClassName='users-container__header-content__button greenButton btn btn-success'
            buttonOnclick={handleAddUser}
            access={access}
          />
        </div>

        <div className='users-container__table-content '>
          <div className='users-container__table'>
            <DynamicDataTable
              tableHeading=''
              isSearchIconEnable={true}
              isFilterIconEnable={true}
              columns={columns}
              data={users}
              searchValue={searchValue}
              filterValues={filterValues}
              onViewHandler={onViewHandler}
              onEditHandler={onEditHandler}
              onDeleteHandler={onDeleteHandler}
              defaultItemsPerPage={itemsPerPage}
              errorMessage={errorMessage}
              isEditIconEnable={true}
              isDeleteIconEnable={true}
              access={access}
              rowsPerPageValueChange={rowsPerPageValueChange}
            />
          </div>
        </div>
      </div>
    </>
  );
};

export default UserManagementPage;
