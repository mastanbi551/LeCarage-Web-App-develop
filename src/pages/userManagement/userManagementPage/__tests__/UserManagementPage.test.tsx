import React from 'react';
import { render, screen } from '@testing-library/react';
import UserManagementPage from '../UserManagementPage';

describe('UserManagementPage', () => {
  const handleAddUserMock = jest.fn();
  const onViewHandlerMock = jest.fn();
  const onEditHandlerMock = jest.fn();
  const onDeleteHandlerMock = jest.fn();


  const columns = [
    
    { Header: 'User Type', accessor: 'User Type' },
    { Header: 'First Name', accessor: 'First Names' },
    { Header: 'Last Name', accessor: 'Last Name' },
    { Header: 'Email Address', accessor: 'Email Address' },
    { Header: 'CellPhone Number', accessor: 'CellPhone Number' },
    { Header: 'Status', accessor: 'Status' },
  ];

  const users = [
    {
      'CellPhone Number': '1234567890',
      'Email Address': 'testnew@example.com',
      'First Name': 'Doe',
      'Last Name': 'John',
      'Status': 'Active',
      'User Id': 'doejohn123',
      'User Type': 'Regular',
      'Receptionist': 'Yes',
      'Prathima':'prat'
    },
    {
      'CellPhone Number': '123-456-7890',
      'Email Address': 'test@example.com',
      'First Name': 'John',
      'Last Name': 'Doe',
      'Status': 'Inactive',
      'User Id': 'johndoe123',
      'User Type': 'Regular',
      'Receptionist': 'Yes',
      'Prathima':'Pti'
    }
  ];

  const searchValue = ['Admin'];
  const filterValues = [{ mainMenu: 'Filter', subMenus: ['Admin'] }];
  const errorMessage = 'Error message';

  it('should match the snapshot', () => {
    const { container } = render(
      <UserManagementPage
        handleAddUser={handleAddUserMock}
        columns={columns}
        searchValue={searchValue}
        onViewHandler={onViewHandlerMock}
        onEditHandler={onEditHandlerMock}
        onDeleteHandler={onDeleteHandlerMock}
        users={users}
        filterValues={filterValues}
        errorMessage={errorMessage}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call handleAddUsertype when Add New button is clicked', () => {
    render(
      <UserManagementPage
        handleAddUser={handleAddUserMock}
        columns={columns}
        searchValue={searchValue}
        onViewHandler={onViewHandlerMock}
        onEditHandler={onEditHandlerMock}
        onDeleteHandler={onDeleteHandlerMock}
        users={users}
        filterValues={filterValues}
        errorMessage={errorMessage}
      />
    );

    const addButton = screen.getByRole('button', { name: 'Add New' });
    addButton.click();

    expect(handleAddUserMock).toHaveBeenCalled();
  });

 

});