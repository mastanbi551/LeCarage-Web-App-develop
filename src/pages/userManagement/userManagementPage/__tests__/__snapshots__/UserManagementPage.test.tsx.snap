// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserManagementPage should match the snapshot 1`] = `
<div>
  <div
    class="users-container"
  >
    <div
      class="users-container__header-content"
    >
      <h1
        class="users-container__header-content__heading"
      >
        Users
      </h1>
      <button
        class="users-container__header-content__button greenButton btn btn-success"
        type="button"
      >
        Add New
      </button>
    </div>
    <div
      class="users-container__table-content "
    >
      <div
        class="users-container__table"
      >
        <div
          class="text-align-center norecord"
        >
          Error message
        </div>
      </div>
    </div>
  </div>
</div>
`;
